.github/workflows/reusable-release.yml

```diff
@@ -342,25 +343,32 @@ jobs:
         run: |
           echo "KEY_FINGERPRINT=$KEY_FINGERPRINT"
           find $LOCAL_ARTIFACTS_STAGING_PATH -type f -exec gpg -a --local-user "$KEY_FINGERPRINT" --detach-sign {} \;
+      - name: "Full Main-Branch release: Add release commit (from temporary release branch) to default branch"
+        if: needs.init.outputs.release_type == 'FULL_MAIN_BRANCH'
+        env:
+          GH_TOKEN: ${{ steps.generate-github-app-token.outputs.token }}
+          GH_REPO: ${{ github.repository }}
+          RELEASE_COMMIT_ID: ${{ needs.push-release-commit.outputs.release_commit_id }}
+        run: |
+          if gh api --silent --method PATCH /repos/:owner/:repo/git/refs/heads/$GITHUB_REF_NAME -f "sha=$RELEASE_COMMIT_ID"; then
+            echo "...fast-forward of default branch to include release commit succeeded"
+          else
+            echo "...fast-forward failed (commits added to default branch while release running?), will attempt a merge instead"
+            gh api --silent --method POST /repos/:owner/:repo/merges -f "base=$GITHUB_REF_NAME" -f "head=$RELEASE_COMMIT_ID"
```

Comment:
[`POST /repos/:owner/:repo/merges`](https://docs.github.com/en/rest/branches/branches?apiVersion=2022-11-28#merge-a-branch) is the GitHub API endpoint for merging a branch (that isn't necessarily a PR branch).